### PR TITLE
Remove leftover chalk types

### PR DESCRIPTION
**Pull request by jackh726**
Split out from #69406

Since the other PR is having memory problems with `parallel-compiler = true`, figured I should split this out. Surprisingly, this actually changes some errors, and I'm not quite sure why.

r? @nikomatsakis

### CODE REVIEW

**Review by nikomatsakis — APPROVED** [review]

**Comment by nikomatsakis** [comment]
@bors delegate=jackh725

I approve, but we should wait until CI is happy. @jackh726 you should be able to r+

**Comment by jackh726** [comment]
@nikomatsakis Are the error changes okay? Do you know *why* they changed?

**Comment by nikomatsakis** [comment]
lol, so I did

@bors delegate=jackh726

File: src/librustc_infer/infer/nll_relate/mod.rs [hunk 1]
```diff
@@ -279,12 +273,7 @@ where
                 var
             }
 
-            _ => {
-                let projection = ty::ProjectionPredicate { projection_ty, ty: value_ty };
-                self.delegate
-                    .push_domain_goal(DomainGoal::Holds(WhereClause::ProjectionEq(projection)));
-                value_ty
-            }
+            _ => value_ty,
```
Comment (nikomatsakis) [inline src/librustc_infer/infer/nll_relate/mod.rs:276]: ok so maybe I r+'d a bit too hastily -- it seems like we ought to be assering that this path is unreachable, no?

**Review by nikomatsakis — COMMENTED** [review]
It's true, I don't know what's causing the errors -- skimming the diff, I only found one thing that might be a problem.

File: src/librustc_infer/infer/nll_relate/mod.rs (hunk 1 above)
Comment (jackh726) [reply src/librustc_infer/infer/nll_relate/mod.rs:276]: Hmm, I can check

Comment (jackh726) [reply src/librustc_infer/infer/nll_relate/mod.rs]: I added back the same assertion that would have been hit before.

**Comment by jackh726** [comment]
Looks like overflow error changes might be related to `parallel-compiler = true` See #67362 and #67379

I'm gonna try to run test locally with `parallel-compiler = false`

**Comment by jackh726** [comment]
Indeed, setting `parallel-compiler = false` makes the errors be as they are on master. Gonna wait for CI to finish to see what it says. I'm curious if it'll be different between mingw-check and x86_64-gnu-llvm-8.

**Pull request merged**